### PR TITLE
improvement: improve loading states while islands are initializing

### PR DESCRIPTION
**Pull request by mscolnick**
Various improvements for islands:
* improve loading states while islands are initializing
* fix some flickering CSS when loading
* fix newlines in markdown rendering on first load
* stylings fixes for islands height/padding
* create a namespace `islands`

### CODE REVIEW

File: frontend/src/core/islands/bridge.ts [hunk 1]
```diff
@@ -169,3 +171,11 @@ export class IslandsPyodideBridge implements RunRequests, EditRequests {
     });
   }
 }
+
+function makeRelativeWorkerUrl(url: string) {
+  return url.startsWith("./")
```
Comment (dmadisetti) [inline frontend/src/core/islands/bridge.ts:176]: Normally I test islands behind a proxy, and this isn't always consistent. Will try

File: marimo/_islands/island_generator.py [hunk 2]
```diff
@@ -397,12 +398,12 @@ def render_init_island(self) -> str:
             <marimo-island
                 data-app-id="{self._app_id}"
                 data-cell-id="{init_cell_id}"
-                data-reactive="{json.dumps(True)}"
+                data-reactive="{json.dumps(False)}"
             >
                 <marimo-cell-output>
                 {init_output}
                 </marimo-cell-output>
-                {init_input}
```
Comment (dmadisetti) [inline marimo/_islands/island_generator.py:405]: Was this just a NameError before? Maybe a unit test to catch this branch

Comment (dmadisetti) [reply marimo/_islands/island_generator.py:405]: But also, is init_island even needed with the toast?

**Review by dmadisetti — COMMENTED** [review]
Reads well to me! I haven't tried it out, but eager for the toast !

Is the spinning loader still needed with the toast?

File: frontend/src/core/islands/bridge.ts (hunk 1 above)
Comment (mscolnick) [reply frontend/src/core/islands/bridge.ts:176]: depends what i am testing - but mostly ill just run:
`pnpm dev:islands` and then access the server with `<script type="module" src="/src/core/islands/main.ts"></script>`

File: marimo/_islands/island_generator.py (hunk 2 above)
Comment (mscolnick) [reply marimo/_islands/island_generator.py:405]: > Was this just a NameError before? Maybe a unit test to catch this branch

It wasn't a name error. there was no error, I just inlined the variable. ill create a snapshot test

> is init_island even needed with the toast

they somewhat have different use-cases. `init_island` shows a spinner while HTML loads (seems like you may want to show before any static HTML appears). the toast happens only when you interact with static HTML before it "rehydrates"

**Pull request merged**